sdk/samples/iot/CMakeLists.txt

```diff
@@ -1,38 +1,53 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # SPDX-License-Identifier: MIT
 
+if (TRANSPORT_PAHO)
+
 cmake_minimum_required (VERSION 3.10)
 
-project (az_iot_samples LANGUAGES C)
+project (az_iot_samples_common LANGUAGES C)
 
 set(CMAKE_C_STANDARD 99)
 
+find_package(eclipse-paho-mqtt-c CONFIG)
+if(NOT eclipse-paho-mqtt-c_FOUND)
+  find_package(eclipse-paho-mqtt-c REQUIRED)
+endif()
+
 find_package(OpenSSL CONFIG)
 if(NOT OpenSSL_FOUND)
   find_package(OpenSSL REQUIRED)
 endif()
 
+add_subdirectory(hub)
+add_subdirectory(provisioning)
```

Comment:
sorry to nit here. I think these should go at the end after this lib is built. From a logical standpoint, this "common" stuff should be built up before adding other things which depend on it. I think technically CMake allows it but it's better flow in my head.